src/components/EditActivityPage/EditActivityPage.js

```diff
@@ -35,17 +35,17 @@ const EditActivityPage = props => (
 EditActivityPage.propTypes = {
   activity: PropTypes.shape(activityPropTypeShape).isRequired,
   history: PropTypes.shape(historyPushPropTypeShape).isRequired,
-  onEditActivity: PropTypes.func.isRequired,
-  onRemoveActivity: PropTypes.func.isRequired,
+  onStartEditActivity: PropTypes.func.isRequired,
+  onStartRemoveActivity: PropTypes.func.isRequired,
 };
 
 const mapStateToProps = (state, props) => ({
   activity: getEditedActivity(state, props),
 });
 
 const mapDispatchToProps = {
-  onEditActivity: editActivity,
-  onRemoveActivity: removeActivity,
+  onStartEditActivity: startEditActivity,
```

Comment:
in general, I don't like this `onStartEditActivity`. 

You could try to handle it in prettier way. There is a nice package to handle requests `redux-axios-middleware`. 
It will be a good job if you will handle it using only documentation. Do it in separate PR.